packages/workbox-core/test/bundle/node/utils/test-logHelper.js

```diff
@@ -0,0 +1,339 @@
+import chai from 'chai';
+import sinon from 'sinon';
+
+import logHelper from '../../../../src/utils/logHelper';
+
+const expect = chai.expect;
+
+describe(`logHelper [${process.env.NODE_ENV}]`, function() {
+  let stubs = [];
+
+  beforeEach(function() {
+    logHelper.logLevel = logHelper.LOG_LEVELS.verbose;
+  });
+
+  afterEach(function() {
+    stubs.forEach((stub) => {
+      stub.restore();
+    });
+
+    stubs = [];
+  });
+
+  describe(`LOG_LEVELS`, function() {
+    it(`should expose the valid LOG_LEVELS`, function() {
+      expect(logHelper.LOG_LEVELS).to.exist;
+    });
+
+    it(`should expose the expected LOG_LEVELS`, function() {
+      expect(logHelper.LOG_LEVELS.verbose).to.exist;
+      expect(logHelper.LOG_LEVELS.debug).to.exist;
+      expect(logHelper.LOG_LEVELS.warning).to.exist;
+      expect(logHelper.LOG_LEVELS.error).to.exist;
+    });
+  });
+
+  describe(`.logLevel (setter)`, function() {
+    it(`should allow valid log levels`, function() {
+      expect(() => {
+        logHelper.logLevel = logHelper.LOG_LEVELS.verbose;
+        logHelper.logLevel = logHelper.LOG_LEVELS.debug;
+        logHelper.logLevel = logHelper.LOG_LEVELS.warning;
+        logHelper.logLevel = logHelper.LOG_LEVELS.error;
+      }).to.not.throw();
+    });
+
+    // TODO: Catch WorkboxError with error code
+    it(`should not allow log level less than verbose`, function() {
+      expect(() => {
+        logHelper.logLevel = logHelper.LOG_LEVELS.verbose - 1;
+      }).to.throw();
+    });
+
+    // TODO: Catch WorkboxError with error code
+    it(`should not allow log level greater than error`, function() {
+      expect(() => {
+        logHelper.logLevel = logHelper.LOG_LEVELS.error + 1;
+      }).to.throw();
+    });
+
+    // TODO: Catch WorkboxError with error code
+    it(`should not allow non-number log levels`, function() {
+      expect(() => {
```

Comment:
There's a bunch of places in the tests where we pass in a variety of bad inputs, and expect each to fail. It would be nice to standardize on something that made it easier to write those test cases, but also made it clear when there's a failure, what the underlying input was that caused the failure. (The way this test is written, there's no indication that 4 of the expects might have passed, but the one with `null` failed.)

Ideally, this would be done in some sort of registerTestCasesFromBadInput() function that took in a description, a set of bad input, and the logic that should be applied to each bad input, and then in our tests we could do:

```js
registerTestCasesFromBadInput(
  'should not allow non-number log levels',
  [undefined, null, '', [], {}],
  (badInput) => logHelper.logLevel = badInput
);
```

And then, under the hood, it would register a test case for each `badInput`:

```js
it(`should not allow non-number log levels (test input: ${badInput})`, function() {
  // logic goes here
});
```

I don't want to write the actual logic for `registerTestCasesFromBadInput()` within a comment in a PR, but we could leave that as a TODO or something.